relay-server/src/actors/upstream.rs

```diff
@@ -133,6 +157,47 @@ impl UpstreamRateLimits {
     }
 }
 
+/// Requests are queued and send to the HTTP connections according to their priorities
+/// High priority messages are sent first and then, when no high priority message is pending,
+/// low priority messages are sent. Within the same priority messages are sent FIFO.
+pub enum RequestPriority {
+    /// High priority, low volume messages (e.g. ProjectConfig, ProjectStates, Registration messages)
+    High,
+    /// Low priority, high volume messages (e.g. Events and Outcomes)
+    Low,
+}
+
+/// An type implementing this trait can be placed in a queue according to its priority.
+/// This trait is implemented by message types that need to be sent via http
+pub trait WithRequestPriority {
+    fn priority() -> RequestPriority;
```

Comment:
I think we can move this function into `UpstreamQuery`, as this is the only place where this functionality is needed. That saves us from implementing a separate trait. 

Also: Consider to add a default implementation that returns `RequestPriority::Low`. This is very explicit in places where we need high prio.